.github/workflows/check.yaml

```diff
@@ -0,0 +1,37 @@
+name: Run static analysis
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  run-static-analysis:
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request' &&
+      github.event.pull_request.head.repo.full_name != github.repository
+
+    runs-on: ubuntu-22.04
+
+    steps:
+    - uses: actions/checkout@master
+
+    - name: Setup Tarantool CE
+      uses: tarantool/setup-tarantool@v3
+      with:
+        tarantool-version: '3.2.0'
+
+    - name: Setup tt
+      run: |
+        curl -L https://tarantool.io/release/3/installer.sh | sudo bash
+        sudo apt install -y tt
+        tt version
+
+    - name: Setup luacheck
+      run: |
+        tt rocks install luacheck 0.26.0
```

Comment:
If would be nice to separate Makefile lint deps and test deps. But it's up to you. You also can see this example: https://github.com/tarantool/sideservice/blob/master/Makefile

Reply:
Decompose an installation of deps in a separate commit.